src/compiler/program.ts

```diff
@@ -626,6 +631,17 @@ namespace ts {
 
         Debug.assert(!!missingFilePaths);
 
+        // List of collected files is complete; validate exhautiveness if this is a project with a file list
+        if (options.composite && rootNames.length < files.length) {

```

Comment:
we have done most of these project validation type checks in `verifyCompilerOptions`